leadbutt.py

```diff
@@ -133,18 +135,34 @@ def leadbutt(config_file, cli_options, verbose=False, **kwargs):
         end_time = datetime.datetime.utcnow()
         start_time = end_time - datetime.timedelta(
             seconds=period_local * count_local)
-        results = conn.get_metric_statistics(
-            period_local,  # minimum: 60
-            start_time,
-            end_time,
-            metric['MetricName'],  # RequestCount, CPUUtilization
-            metric['Namespace'],  # AWS/ELB, AWS/EC2
-            metric['Statistics'],  # Sum, Maximum
-            dimensions=metric['Dimensions'],
-            unit=metric['Unit'],  # Count, Percent
-        )
-        # sys.stderr.write('{} {}\n'.format(options['Count'], len(results)))
-        output_results(results, metric, options)
+        # if the unit is in the yaml config, send it and get only those.
+        # else don't send it, get all the available units, and in output_results,
+        # we'll loop over all the results and send all to graphite.
+        if 'Unit' in metric.keys():
+            unit = metric['Unit']
+        else:
+            unit = None
+        # make sure that if we get one metric name, we make it a list and loop over
+        # the one value
+        metric_names = metric['MetricName']
```

Comment:
likewise, I think you can get rid of this comment, and if you re-write the casting code like:
```
metric_names = metric['MetricName']
if not isinstance(metric_names, list):
    metric_names = [metric_names]
```
it's more obvious what's happening